agents/web-api/src/main/java/edu/jhuapl/dorset/agents/StockAgent.java

```diff
@@ -0,0 +1,274 @@
+                                        + keywordCompanyName + ".").replace(
+                                        "..", ".")));
+            }
+            return new AgentResponse(ResponseStatus.Code.AGENT_DID_NOT_KNOW_ANSWER);
+        }
+        
+        JsonObject returnObj = processData(json, keywordCompanyName);
+
+        // replace ".." with "." to maintain proper grammar when the
+        // keyword contains an abbreviation
+        if (returnObj != null) {
+            return new AgentResponse(Response.Type.JSON,
+                    ("Here is the longitudinal stock market data from the last "
+                            + DAYS_IN_A_MONTH
+                            + " days for "
+                            + keywordCompanyName + ".").replace("..", "."),
+                    returnObj.toString());
+        }
+        return new AgentResponse(ResponseStatus.Code.AGENT_DID_NOT_KNOW_ANSWER);
+
+    }
+
+    protected CompanyInfo findStockSymbol(String stockCompanyName) {
+        CompanyInfo companyInfo = null;
+        ArrayList<String> regexMatches = new ArrayList<String>();
+
+        if (this.stockSymbolMap.get(stockCompanyName) != null) {
+            companyInfo = this.stockSymbolMap.get(stockCompanyName);
+
+        } else {
+            String regex = "\\b" + stockCompanyName + "\\b";
+
+            Pattern pat = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
+
+            for (Map.Entry<String, CompanyInfo> entry : stockSymbolMap
+                    .entrySet()) {
+                Matcher matcher = pat.matcher(entry.getKey());
+
+                if (matcher.find()) {
+                    regexMatches.add(entry.getKey());
+                }
+            }
+
+            if (regexMatches.size() == 0) {
+                companyInfo = null;
+            } else if (regexMatches.size() == 1) {
+                companyInfo = this.stockSymbolMap.get(regexMatches.get(0));
+
+            } else {
+                int distance;
+                HashMap<String, Integer> matchDistanceMap = new HashMap<String, Integer>();
+                for (int i = 0; i < regexMatches.size(); i++) {
+                    distance = (StringUtils.getLevenshteinDistance(regexMatches.get(i), stockCompanyName));
+                    matchDistanceMap.put(regexMatches.get(i), distance);
+                }
+
+                Entry<String, Integer> minDistancePair = null;
+                for (Entry<String, Integer> entry : matchDistanceMap.entrySet()) {
+                    if (minDistancePair == null || minDistancePair.getValue() > entry.getValue()) {
+                        minDistancePair = entry;
+                    }
+                }
+
+                companyInfo = this.stockSymbolMap.get(minDistancePair.getKey());
+
+            }
+
+        }
+
+        return companyInfo;
+    }
+
+    protected JsonObject processData(String json, String keyWordCompanyName) {
```

Comment:
Could you describe the format of the data being processed here? One way to do that is to pretty print a json file used for the tests and then point the developer at that file.